cmd/watch.go

```diff
@@ -0,0 +1,214 @@
+			return result.FromError(errors.New("update cancelled"))
+		case res != nil:
+			return PrintEngineResult(res)
+		default:
+			return nil
+		}
+	}
+
+	var cmd = &cobra.Command{
+		Use:        "watch",
+		Aliases:    []string{"watch"},
+		SuggestFor: []string{"developer", "dev"},
+		Short:      "Continuously update the resources in a stack",
+		Long: "Continuously update the resources in a stack.\n" +
+			"\n" +
+			"This command watches the working directory for the current project and updates the active stack whenever\n" +
+			"the project changes.  In parallel, logs are collected for all resources in the stack and displayed along\n" +
+			"with update progress.\n" +
+			"\n" +
+			"The program to watch is loaded from the project in the current directory by default. Use the `-C` or\n" +
+			"`--cwd` flag to use a different directory.",
+		Args: cmdutil.MaximumNArgs(1),
+		Run: cmdutil.RunResultFunc(func(cmd *cobra.Command, args []string) result.Result {
+			interactive := cmdutil.Interactive()
+			if !interactive {
+				yes = true // auto-approve changes, since we cannot prompt.
+			}
+
+			opts, err := updateFlagsToOptions(interactive, skipPreview, yes)
+			if err != nil {
+				return result.FromError(err)
+			}
+
+			opts.Display = display.Options{
+				Color:                cmdutil.GetGlobalColorization(),
+				ShowConfig:           showConfig,
+				ShowReplacementSteps: showReplacementSteps,
+				ShowSameResources:    showSames,
+				SuppressOutputs:      suppressOutputs,
+				IsInteractive:        interactive,
+				Type:                 display.DisplayWatch,
+				Debug:                debug,
+			}
+
+			return upWorkingDirectory(opts)
+		}),
+	}
+
+	cmd.PersistentFlags().BoolVarP(
+		&debug, "debug", "d", false,
+		"Print detailed debugging output during resource operations")
+	cmd.PersistentFlags().BoolVar(
+		&expectNop, "expect-no-changes", false,
+		"Return an error if any changes occur during this update")
+	cmd.PersistentFlags().StringVarP(
+		&stack, "stack", "s", "",
+		"The name of the stack to operate on. Defaults to the current stack")
+	cmd.PersistentFlags().StringVar(
+		&stackConfigFile, "config-file", "",
+		"Use the configuration values in the specified file rather than detecting the file name")
+	cmd.PersistentFlags().StringArrayVarP(
+		&configArray, "config", "c", []string{},
+		"Config to use during the update")
+	cmd.PersistentFlags().BoolVar(
+		&path, "config-path", false,
+		"Config keys contain a path to a property in a map or list to set")
+	cmd.PersistentFlags().StringVar(
+		&secretsProvider, "secrets-provider", "default", "The type of the provider that should be used to encrypt and "+
+			"decrypt secrets (possible choices: default, passphrase, awskms, azurekeyvault, gcpkms, hashivault). Only"+
+			"used when creating a new stack from an existing template")
+
+	cmd.PersistentFlags().StringVarP(
+		&message, "message", "m", "",
+		"Optional message to associate with the update operation")
```

Comment:
How does this work? Every update that happens due to changes will get the same message?

Reply:
Yes - that's currently the way this would work.  For now I think that's okay - long term we may choose to diverge from having every update be a full "update" in the backend.